cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProfileHttpHandler.java

```diff
@@ -161,4 +168,22 @@ public void enableProfile(HttpRequest request, HttpResponder responder,
     // TODO: implement the method
     responder.sendStatus(HttpResponseStatus.OK);
   }
+
+  private void validateProvisionerProperties(ProfileCreateRequest request) throws BadRequestException {
+    ProvisionerInfo provisionerInfo = request.getProvisioner();
+    Map<String, String> properties = new HashMap<>();
+    for (ProvisionerPropertyValue value : provisionerInfo.getProperties()) {
+      properties.put(value.getName(), value.getValue());
+    }
+    try {
+      provisioningService.validateProperties(provisionerInfo.getName(), properties);
+    } catch (NotFoundException e) {
+      throw new BadRequestException(String.format("The specified provisioner %s does not exist, " +
+                                                    "thus cannot be associated with a profile",
+                                                  provisionerInfo.getName()), e);
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException(String.format("Invalid properties in provisioner %s",
```

Comment:
Should include the message from the IllegalArgumentException in the message for the BadRequest, otherwise the user won't know what is invalid.